### PR TITLE
Add spamming detection to inbound SMS conversation

### PR DESCRIPTION
**Pull request by imsickofmaps**
Need to be able to identify users who are spamming the SMS line, with the same message sent more than than 5 times within 10 hours. This will be done by keeping a copy in contact object of last unique message, timestamp first seen and counter of that message. Users who violate spam criteria will be marked as spammer in their extras. We will create a Smart group to add this list of users into.

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@smn ready for review - #24 is merged into this and tweaked

File: js_sandbox/lib/go-nike-ghr-sms.js [hunk 1]
```diff
@@ -171,34 +171,59 @@ function GoNikeGHRSMS() {
...
+                    }
+                }
+                return im.api_request('contacts.update_extras', {
+                    key: contact.key,
+                    fields: fields
                 });
-                p.add_callback(function(result){
+            });
+            p.add_callback(function(result){
+                if (includes_swear || is_spammer){
```
Comment (smn) [inline js_sandbox/lib/go-nike-ghr-sms.js:212]: Does `is_spammer` need to be stored on the contact object?
Does `is_spammer` perhaps require a different response?

**Comment by smn** [comment]
:+1: other than the two questions on `is_spammer` which are largely to be answered by the existing SoW.

File: js_sandbox/lib/go-nike-ghr-sms.js (hunk 1 above)
Comment (imsickofmaps) [inline js_sandbox/lib/go-nike-ghr-sms.js:212]: @smn it does get stored on the contact object - line 199 - at the time they become a spammer. The SoW just requires them to be Smart-group-able ... no response different. But its a simple change if they request. Thanks.

**Comment by smn** [comment]
@imsickofmaps searching for contacts that have some value for `is_spammer` is harder than just searching for `"is_spammer:true"`

**Comment by imsickofmaps** [comment]
@smn I did a datetime so they could do range of dates - spammers this week - etc. But I actually don't know how they intend to use it. I'll ask and change if over-engineered :)

**Pull request merged**